core/templates/pages/email-dashboard-pages/email-dashboard-page.component.ts

```diff
@@ -16,114 +16,140 @@
-
-require('services/user.service.ts');
-
-require('./email-dashboard-data.service');
-require(
-  'components/forms/schema-based-editors/schema-based-editor.directive.ts');
-
-angular.module('oppia').component('emailDashboardPage', {
-  template: require('./email-dashboard-page.component.html'),
-  controller: [
-    '$rootScope', 'EmailDashboardDataService', 'LoaderService',
-    'UserService', 'EMAIL_DASHBOARD_PREDICATE_DEFINITION', function(
-        $rootScope, EmailDashboardDataService, LoaderService,
-        UserService, EMAIL_DASHBOARD_PREDICATE_DEFINITION) {
-      var ctrl = this;
-      ctrl.resetForm = function() {
-        ctrl.data = {};
-        EMAIL_DASHBOARD_PREDICATE_DEFINITION.forEach(predicate => {
-          ctrl.data[predicate.backend_attr] = predicate.default_value;
-        });
-      };
-
-      ctrl.areAllInputsEmpty = function() {
-        return Object.values(ctrl.data).every(
-          value => value === null || value === false);
-      };
-
-      ctrl.submitQueryAsync = async function() {
-        EmailDashboardDataService.submitQueryAsync(ctrl.data).then(
-          function(queries) {
-            ctrl.currentPageOfQueries = queries;
-            // TODO(#8521): Remove the use of $rootScope.$apply()
-            // once the directive is migrated to angular.
-            $rootScope.$apply();
-          });
-        ctrl.resetForm();
-        ctrl.showSuccessMessage = true;
-      };
-
-      ctrl.getNextPageOfQueries = function() {
-        if (EmailDashboardDataService.isNextPageAvailable()) {
-          EmailDashboardDataService.getNextQueriesAsync().then(
-            function(queries) {
-              ctrl.currentPageOfQueries = queries;
-              // TODO(#8521): Remove the use of $rootScope.$apply()
-              // once the directive is migrated to angular.
-              $rootScope.$apply();
-            });
+import { ChangeDetectorRef, Component } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { AppConstants } from 'app.constants';
+import { QueryData } from 'domain/email-dashboard/email-dashboard-backend-api.service';
+import { EmailDashboardQuery } from 'domain/email-dashboard/email-dashboard-query.model';
+import { LoaderService } from 'services/loader.service';
+import { Schema } from 'services/schema-default-value.service';
+import { UserService } from 'services/user.service';
+import { EmailDashboardDataService } from './email-dashboard-data.service';
+
+interface CustomizationArgSpec {
+  'backend_id': string;
+  'backend_attr': string;
+  'description': string;
+  'schema': Schema;
+  'default_value': number | boolean | string;
+}
+
+@Component({
+  selector: 'oppia-email-dashboard-page',
+  templateUrl: './email-dashboard-page.component.html'
+})
+export class EmailDashboardPageComponent {
+  data: QueryData;
+  currentPageOfQueries: EmailDashboardQuery[] = [];
+  showSuccessMessage: boolean = false;
+  username: string = '';
```

Comment:
```suggestion
  data!: QueryData;
  currentPageOfQueries!: EmailDashboardQuery[];
  username!: string;
  showSuccessMessage: boolean = false;
```
We should use lifecycle hooks to initialize these empty variables. If not, leave them undefined.